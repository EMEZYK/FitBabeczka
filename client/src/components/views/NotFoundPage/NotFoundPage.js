import React from "react";
import {
  NotFoundPageContainer,
  NotFoundPageImage,
} from "./NotFoundPage.styled";

const NotFoundPage = () => {
  return (
    <NotFoundPageContainer>
      <NotFoundPageImage alt="not-found-page" src="/photos/notFoundPage.jpg" />
    </NotFoundPageContainer>
  );
};

export default NotFoundPage;
