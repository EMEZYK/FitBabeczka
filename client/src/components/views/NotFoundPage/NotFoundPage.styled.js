import styled from "styled-components";

export const NotFoundPageContainer = styled.div`
  display: flex;
  height: 100vh;
  align-items: center;
  justify-content: center;
  margin: auto 0;
  background: linear-gradient(0.25turn, #3f87a6, #ebf8e1, #c07461);

  @media (max-width: 1024px) {
    background: #fff;
  }
`;

export const NotFoundPageImage = styled.img`
  max-height: 80%;
  width: 85vw;
  object-fit: cover;
  object-position: 20% 69%;

  @media (max-width: 1024px) {
    min-height: 100vh;
    width: 100vw;
  }

  @media (max-width: 991px) {
    object-fit: contain;
    object-position: bottom;
  }
`;
