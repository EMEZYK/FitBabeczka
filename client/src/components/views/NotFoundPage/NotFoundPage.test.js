import React from "react";
import "jest-styled-components";
import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import NotFoundPage from "./NotFoundPage";
import "@testing-library/jest-dom";

describe("NotFoundPage components", () => {
  afterEach(cleanup);

  test("Should take a snapshot',", () => {
    const { asFragment } = render(<NotFoundPage />);
    expect(asFragment(<NotFoundPage />)).toMatchSnapshot();
  });

  test("Renders image", () => {
    render(<NotFoundPage />);
    const page = screen.queryByAltText("not-found-page");
    expect(page).toHaveAttribute("src", "/photos/notFoundPage.jpg");
  });
});
