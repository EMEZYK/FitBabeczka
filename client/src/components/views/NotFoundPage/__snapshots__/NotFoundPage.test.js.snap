// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage components Should take a snapshot', 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: auto 0;
  background: linear-gradient(0.25turn,#3f87a6,#ebf8e1,#c07461);
}

.c1 {
  max-height: 80%;
  width: 85vw;
  object-fit: cover;
  object-position: 20% 69%;
}

@media (max-width:1024px) {
  .c0 {
    background: #fff;
  }
}

@media (max-width:1024px) {
  .c1 {
    min-height: 100vh;
    width: 100vw;
  }
}

@media (max-width:991px) {
  .c1 {
    object-fit: contain;
    object-position: bottom;
  }
}

<div
    class="c0"
  >
    <img
      alt="not-found-page"
      class="c1"
      src="/photos/notFoundPage.jpg"
    />
  </div>
</DocumentFragment>
`;
